src/pybind/mgr/cephadm/inventory.py

```diff
@@ -517,6 +522,22 @@ def load(self):
                     host, e))
                 pass
 
+    def _stray_host_cache_entry(self, host: str) -> str:
+        # return true if entry is stray and must be removed
+        # need to check for normal host cache entry and device entry
+        # <hostname>   OR   <hostname>.device.<integer>
+        if host not in self.mgr.inventory:
+            # check if it's a device cache entry
+            actual_host = host
+            try:
+                actual_host = '.'.join(host.split('.')[:-2])
+            except Exception:
+                return 'stray'
+            if actual_host in self.mgr.inventory:
+                return 'device'
+            return 'stray'
+        return 'found'
```

Comment:
It would be good to add a comment to clarify how `actual_host` is calculated (and why it could fail). Anyway, is the logic above equivalent to the below code?

```
if host in self.mgr.inventory:
    return 'found'
 # check if it's a device cache entry
 try:
     actual_host = '.'.join(host.split('.')[:-2])
     return 'device' if actual_host in self.mgr.inventory else 'stray'
 except Exception:
     return 'stray'

```

Reply:
added a code comment and reformatted section as you have here